experimental/dispatch_profiler/matmul.py

```diff
@@ -297,235 +368,46 @@ def run_and_save(self, output_dir="."):
     result = np.matmul(lhs_np_array, rhs_np_array)
 
     # Save the input data as np.array for the matmul operation.
-    np.save(os.path.join(output_dir, self.filename_lhs),\
-             np.array(lhs_np_array, dtype = self.dtype_lhs))
-    np.save(os.path.join(output_dir, self.filename_rhs),\
-             np.array(rhs_np_array, dtype = self.dtype_rhs))
+    np.save(self.filepath_lhs, np.array(lhs_np_array, dtype=self.dtype_lhs))
+    np.save(self.filepath_rhs, np.array(rhs_np_array, dtype=self.dtype_rhs))
 
     # Save the expected result as an np.array.
-    np.save(os.path.join(output_dir, self.reference_filename_result),\
-             np.array(result, dtype = self.dtype_result))
-
-
-###############################################################################
-class MatmulOperationLauncher:
-  """Launches the compilation and execution of the matmul operation.
-  MatmulOperationLauncher class has the following responsibilities:
-    1) Launches compilation of the matmul for a verification or profiling runs.
-    2) Launches the verification by running the IREE compiled matmul operation
-       and the python nmpy.matmul.
-    3) Launches the profiling by running the IREE compiled matmul operation.
-  """
+    np.save(self.filepath_reference_result,
+            np.array(result, dtype=self.dtype_result))
 
-  def __init__(self, args, operation):
-    self.operation = operation
+  def is_cached(self):
+    """Checks if the reference run is cached."""
+    if not os.path.exists(self.filepath_lhs) or \
+       not os.path.exists(self.filepath_rhs) or \
+       not os.path.exists(self.filepath_reference_result):
+      return False
+    return True
```

Comment:
```suggestion
    return (os.path.exists(self.filepath_lhs)
      and os.path.exists(self.filepath_rhs)
      and os.path.exists(self.filepath_reference_result))
```

Reply:
If all input and expected file paths exists we return `True` for `ReferenceMatmulOp::is_cached`.

Reply:
Moved the implementation of this to `ReferenceOpInterface` s.t. different operation-specific reference implementations won't have to re-implement this.